pallets/auctions/src/tests/topup.rs

```diff
@@ -0,0 +1,872 @@
+      Error::<Test>::AuctionDoesNotExist,
+    );
+  });
+}
+
+/// Claiming reserved amounts from a TopUp auction
+///
+/// Happy path
+///
+///
+#[test]
+fn claim_topup_auction_without_winner_should_work() {
+  predefined_test_ext().execute_with(|| {
+    let mut common_auction_data = valid_common_auction_data();
+    common_auction_data.reserve_price = Some(1_500);
+
+    let auction = topup_auction_object(common_auction_data, valid_topup_specific_data());
+
+    assert_ok!(AuctionsModule::create(Origin::signed(ALICE), auction));
+
+    set_block_number::<Test>(11);
+
+    assert_ok!(AuctionsModule::bid(
+      Origin::signed(BOB),
+      0,
+      BalanceOf::<Test>::from(1_000_u32)
+    ));
+
+    set_block_number::<Test>(12);
+
+    assert_ok!(AuctionsModule::bid(
+      Origin::signed(CHARLIE),
+      0,
+      BalanceOf::<Test>::from(1_100_u32)
+    ));
+
+    set_block_number::<Test>(22);
+
+    assert_ok!(AuctionsModule::close(Origin::signed(ALICE), 0));
+
+    // Bob claims
+    let bob_balance_before = Balances::free_balance(&BOB);
+    let auction_subaccount_balance_before = Balances::free_balance(&get_auction_subaccount_id(0));
+
+    assert_ok!(AuctionsModule::claim(Origin::signed(BOB), BOB, 0));
+
+    let bob_balance_after = Balances::free_balance(&BOB);
+    let auction_subaccount_balance_after = Balances::free_balance(&get_auction_subaccount_id(0));
+
+    assert_eq!(bob_balance_before.saturating_add(1000), bob_balance_after);
+    assert_eq!(
+      auction_subaccount_balance_before.saturating_sub(1000),
+      auction_subaccount_balance_after
+    );
+
+    // Bob cannot claim twice
+    let bob_balance_before = Balances::free_balance(&BOB);
+    let auction_subaccount_balance_before = Balances::free_balance(&get_auction_subaccount_id(0));
+
+    assert_noop!(
+      AuctionsModule::claim(Origin::signed(BOB), BOB, 0),
+      Error::<Test>::NoReservedAmountAvailableToClaim
+    );
+
+    let bob_balance_after = Balances::free_balance(&BOB);
+    let auction_subaccount_balance_after = Balances::free_balance(&get_auction_subaccount_id(0));
+
+    assert_eq!(bob_balance_before, bob_balance_after);
+    assert_eq!(auction_subaccount_balance_before, auction_subaccount_balance_after);
+
+    // Bob claims for Charlie
+    let charlie_balance_before = Balances::free_balance(&CHARLIE);
+
+    assert_ok!(AuctionsModule::claim(Origin::signed(BOB), CHARLIE, 0));
+
+    let charlie_balance_after = Balances::free_balance(&CHARLIE);
+    assert_eq!(charlie_balance_before.saturating_add(1100), charlie_balance_after);
+
+    let final_auction_account_balance = Balances::free_balance(&get_auction_subaccount_id(0));
+    assert_eq!(final_auction_account_balance, Zero::zero());
+
+    // Auction data is destroyed
+    assert!(matches!(AuctionsModule::auctions(0), None));
+    assert!(matches!(AuctionsModule::auction_owner_by_id(0), None));
+  });
+}
+
+#[test]
+fn claim_topup_auction_with_winner_should_not_work() {
```

Comment:
I'm confused about this test. AFAICT this behaviour leads to Bob's funds being locked forever?

Reply:
https://app.clickup.com/t/2np0jxq